tests/senml_phydat/main.c

```diff
@@ -0,0 +1,290 @@
+        .senml1 = senml_f(24, SENML_UNIT_HOUR),
+        .senml2 = senml_df(24, 0, SENML_UNIT_HOUR),
+        .dim = 2
+    },
+    {
+        .phydat = { { 0, 0, 0 }, UNIT_TEMP_F, 3 },
+        .senml1 = senml_f(255.37, SENML_UNIT_KELVIN),
+        .senml2 = senml_df(0, 3, SENML_UNIT_NONE),
+    },
+    {
+        .phydat = { { 314, 0, 0 }, UNIT_G, -2 },
+        .senml1 = senml_f(30.792881, SENML_UNIT_METER_PER_SQUARE_SECOND),
+        .senml2 = senml_df(314, -2, SENML_UNIT_NONE),
+    },
+    {
+        .phydat = { { 988, 0, 0 }, UNIT_BAR, -3 },
+        .senml1 = senml_f(98.8e3, SENML_UNIT_PASCAL),
+        .senml2 = senml_df(988, 2, SENML_UNIT_PASCAL),
+    },
+    {
+        .phydat = { { 193, 0, 0 }, UNIT_GPM3, 5 },
+        .senml1 = senml_f(19.3e3, SENML_UNIT_KILOGRAM_PER_CUBIC_METER),
+        .senml2 = senml_df(193, 2, SENML_UNIT_KILOGRAM_PER_CUBIC_METER),
+    },
+    {
+        .phydat = { { 220, 0, 0 }, UNIT_GS, 3 },
+        .senml1 = senml_f(22, SENML_UNIT_TESLA),
+        .senml2 = senml_df(220, -1, SENML_UNIT_TESLA),
+    }
+};
+
+void test_phydat_date_to_senml(void)
+{
+    senml_value_t res;
+
+    for (size_t i = 0; i < ARRAY_SIZE(date_tests); i++) {
+        senml_value_t *exp = &(date_tests[i].senml);
+        phydat_t *dat = &(date_tests[i].phydat);
+
+        phydat_date_to_senml(&res, dat);
+
+        DEBUG("Date: %04" PRIi16 "-%02" PRIi16 "-%02" PRIi16 ": %" PRIi64 "\n",
+              dat->val[2], dat->val[1], dat->val[0], res.value.value.i);
+
+        TEST_ASSERT_EQUAL_INT(exp->attr.unit, res.attr.unit);
+        TEST_ASSERT_EQUAL_INT(exp->value.type, res.value.type);
+        TEST_ASSERT_EQUAL_INT(exp->value.value.i, res.value.value.i);
+    }
+}
+
+void test_phydat_time_to_senml(void)
+{
+    senml_value_t res;
+
+    for (size_t i = 0; i < ARRAY_SIZE(time_tests); i++) {
+        senml_value_t *exp = &(time_tests[i].senml);
+        phydat_t *dat = &(time_tests[i].phydat);
+
+        phydat_time_to_senml(&res, &(time_tests[i].phydat));
+
+        DEBUG("Time: %02" PRIi16 ":%02" PRIi16 ":%02" PRIi16 ": %" PRIi64 " s\n",
+              dat->val[2], dat->val[1], dat->val[0], res.value.value.i);
+
+        TEST_ASSERT_EQUAL_INT(exp->attr.unit, res.attr.unit);
+        TEST_ASSERT_EQUAL_INT(exp->value.type, res.value.type);
+        TEST_ASSERT_EQUAL_INT(exp->value.value.i, res.value.value.i);
+
+    }
+}
+
+void test_phydat_to_senml_float(void)
+{
+    senml_value_t res;
+
+    for (size_t i = 0; i < ARRAY_SIZE(value_tests); i++) {
+        senml_value_t *exp = &(value_tests[i].senml1);
+
+        phydat_to_senml_float(&res, &(value_tests[i].phydat), value_tests[i].dim);
+
+        DEBUG("Float: %" PRIi16 "e%" PRIi16 " %s -> %.f %s\n",
+              value_tests[i].phydat.val[value_tests[i].dim], value_tests[i].phydat.scale,
+              phydat_unit_to_str(value_tests[i].phydat.unit),
+              res.value.value.f,
+              senml_unit_to_str(res.attr.unit));
```

Comment:
```suggestion
        if (IS_ACTIVE(ENABLE_DEBUG)) {
            print_str("Float: ");
            print_s32_dec(value_tests[i].phydat.val[value_tests[i].dim]);
            print_str("e");
            print_s32_dec(value_tests[i].phydat.scale);
            print_str(" ");
            print_str(phydat_unit_to_str(value_tests[i].phydat.unit));
            print_str(" -> ");
            print_float(res.value.value.f, 7);
            print_str(" ");
            print_str(phydat_unit_to_str(res.attr.unit));
            print_str("\n");
        }
```

Debugging I remembered non native can't print float, so I did this locally to investigate.

Reply:
Note that the pseudo module `printf_float` unlocks printing of floats. At least on AVR that works as advertised (or did so when I last tested).